modules/ROOT/pages/database/database-connector-examples.adoc

```diff
@@ -0,0 +1,737 @@
+
+=== Stream in Mule 4 versus Stream in Mule 3
+
+In Mule 3.x, streaming was something you had to specifically enable because it was disabled by default. In Mule 4, streaming is transparent and always enabled. You don’t have to worry about it anymore. You can simply trust that the feature is there.
+
+Another improvement from Mule 3 is that you can now use the new repeatable streams mechanism from Mule 4. This means that streams are now repeatable, and you can have DataWeave and other components process the same stream many times, even in parallel. For more information on repeatable streams, see xref:4.1@mule-runtime::streaming-about.adoc[Streaming in Mule 4.0].
+
+=== Limit Results
+
+Mule runtime allows the connector to handle streaming gracefully. However, that does not mean that it's a good idea to move large chunks of data from the database to Mule. Even with streaming, a typical SQL query can return many rows, each one containing a lot of information. This generates:
+
+* Large network overhead
+* A large resources footprint on the database server
+* Large memory and requirements in the Mule Runtime
+
+The Select operation provides parameters (`fetchSize` and `maxRows`) to help with this:
+
+[source,xml,linenums]
+----
+<db:select fetchSize="200" maxRows="1000" config-ref="dbConfig">
+  <db:sql>select * from some_table</db:sql>
+</db:select>
+----
+
+This example tells the connector not to fetch more that `1000` total rows as the result of an SQL `SELECT` query, but to try to only stream results from the database to the Database connector `200` at a time. So in this example, it would require 5 separate network round trips to retrieve 1000 matching rows.
+
+The combination limits the total amount of information that will be retrieved (the `maxRows` value) and guarantees that the data is returned from the database over the network in smaller chunks (the `fetchSize` value).
+
+[NOTE]
+The `fetchSize` is a hint to the JDBC driver and need not always be enforced. The behavior varies between JDBC driver providers. Often, the default `fetchSize` for a JDBC driver is `10`.
+
+
+== Query Timeout Reference
+
+Sometimes database queries take a long time to execute. The following factors often cause delays in query execution:
+
+* An inefficient query, such as one having improper indexing that iterates over many rows
+* A busy RDBMS or network
+* A lock contention
+
+Generally, it's recommended to set a timeout on the query. To manage timeouts, configure `queryTimeout` and `queryTimeoutUnit`. The following example shows how to set a timeout for the Select operation, but all operations support setting a timeout:
+
+[source,xml,linenums]
+----
+<db:select queryTimeout="0" queryTimeoutUnit="SECONDS" config-ref="dbConfig">
+   <db:sql>select * from some_table</db:sql>
+</db:select>
+----
+
+= Examples: Insert, Update, and Delete Data with the Database Connector - Mule 4
```

Comment:
Check your heading levels throughout. This looks like it's at the same level as the Example intro. I think you added this info to this section. Did you decide that it needed to be under one header as opposed to making these examples part of a longer list?

Reply:
I have not touched the heading levels. This particular topic had many h2 examples underneath it. Should each example topic be h2? Is h3 supported?